src/tree/StaticWebAppTreeItem.ts

```diff
@@ -96,6 +97,16 @@ export class StaticWebAppTreeItem extends AzExtParentTreeItem implements IAzureR
             const deleteResponse = await client.staticSites.deleteStaticSite(this.resourceGroup, this.name);
             await pollAzureAsyncOperation(context, deleteResponse, this.subscription);
 
+            const resourceClient = await createResourceClient([context, this]);
+            const resources: ResourceManagementModels.ResourceListResult = await resourceClient.resources.listByResourceGroup(this.resourceGroup);
+
+            // Recently deleted SWA is returned in the list of resources in the RG, even though it's been deleted.
+            // Only delete if 0 resources, or exactly 1 resource and if the resource is the SWA that has just been deleted.
+            if ((resources.length === 0 || (resources.length === 1 && resources[0].id === this.data.id)) && !resources.nextLink) {
```

Comment:
Rather than doing this check, we could have a background check running to see if `resources.length === 0` for a few minutes.  (Of course, if `resources.length > 1` we don't have to bother checking).  Definitely not a blocker, but this id check is a bit janky

Reply:
What if that background check takes 2 to 3 or even 5 minutes? It might get confusing for the user if it doesn't get deleted soon(ish) after they delete their SWA.

Reply:
Well, typically users wouldn't even see that they have a rg to begin with since we don't explicitly list it anywhere.  We also don't mention that we are clearing any empty resource groups so I don't think a user would have that expectation.  

We could prompt the user if they want to delete the rg if the SWA is the only resource we detect during the delete as well?  We do that for app service plans (if we delete the last app in it) for cost reasons

I think we should still show that the SWA has deleted as soon as that async request reports that it is though.

Reply:
Ok, sounds like we should have some more discussion on this before merging this PR. I'd get input (PM magic) from @fiveisprime on this too.

Reply:
Ok I've simplified the logic to where I'm much happier with the solution.

Before deleting the SWA, I get a list of resources in the resource group. Then after deleting the SWA I delete the resource group if it had only a single resource in it (the SWA). Basically the same as before but the code is less janky.